cardano-cli/src/Cardano/CLI/Json/Friendly.hs

```diff
@@ -358,18 +358,44 @@ friendlyCertificate sbe =
-      ConwayCertificate _ (Ledger.ConwayTxCertDeleg Ledger.ConwayDelegCert{}) ->  "Stake address delegation" .= Aeson.String "Conway TODO"
-      ConwayCertificate _ (Ledger.ConwayTxCertDeleg Ledger.ConwayRegDelegCert{}) -> "Stake address registration and delegation" .= Aeson.String "Conway TODO"
-      ConwayCertificate _ (Ledger.ConwayTxCertPool Ledger.RegPool{}) -> "Pool registration" .= Aeson.String "Conway TODO"
-      ConwayCertificate _ (Ledger.ConwayTxCertPool Ledger.RetirePool{}) -> "Pool retirement" .= Aeson.String "Conway TODO"
-
+      ConwayCertificate _ (Ledger.ConwayTxCertCommittee (Ledger.ConwayRegDRep credential coin)) ->
+        "Drep registration certificate" .= object ["amount" .= coin, "certificate" .= toObject' credential]
+      ConwayCertificate _ (Ledger.ConwayTxCertCommittee (Ledger.ConwayUnRegDRep credential coin)) ->
+        "Drep registration certificate" .= object ["amount" .= coin, "certificate" .= toObject' credential]
+      ConwayCertificate _ (Ledger.ConwayTxCertCommittee (Ledger.ConwayAuthCommitteeHotKey (Shelley.KeyHash coldKey) (Shelley.KeyHash hotKey))) ->
+        "Constitutional committee member hot key registration" .= object ["cold key hash" .= String (textShow coldKey), "hot key hash" .= String (textShow hotKey)]
+      ConwayCertificate _ (Ledger.ConwayTxCertCommittee (Ledger.ConwayResignCommitteeColdKey (Shelley.KeyHash coldKey))) ->
+        "Constitutional committee cold key resignation" .= object ["cold key hash" .= String (textShow coldKey)]
+      ConwayCertificate _ (Ledger.ConwayTxCertDeleg (Ledger.ConwayRegCert stakeCredential maybeCoin)) ->
+        "Stake address registration" .= object ["stake credential" .= stakeCredJson sbe stakeCredential, "coin" .= maybeCoin]
+      ConwayCertificate _ (Ledger.ConwayTxCertDeleg (Ledger.ConwayUnRegCert stakeCredential maybeCoin)) ->
+        "Stake address deregistration" .= object ["stake credential" .= stakeCredJson sbe stakeCredential, "coin" .= maybeCoin]
+      ConwayCertificate _ (Ledger.ConwayTxCertDeleg (Ledger.ConwayDelegCert stakeCredential delegatee)) ->
+        "Stake address delegation" .= object ["stake credential" .= stakeCredJson sbe stakeCredential, "delegatee" .= delegateeJson sbe delegatee]
+      ConwayCertificate _ (Ledger.ConwayTxCertDeleg (Ledger.ConwayRegDelegCert stakeCredential delegatee coin)) ->
+        "Stake address registration and delegation" .=
+          object [ "stake credential" .= stakeCredJson sbe stakeCredential
+                 , "delegatee" .= delegateeJson sbe delegatee
+                 , "coin" .= coin
+                 ]
+      ConwayCertificate _ (Ledger.ConwayTxCertPool (Ledger.RegPool poolParams)) ->
+        "Pool registration" .= object ["pool params" .= poolParamsJson sbe poolParams]
+      ConwayCertificate _ (Ledger.ConwayTxCertPool (Ledger.RetirePool (Shelley.KeyHash kh) epoch)) ->
+        "Pool retirement" .= object ["stake pool key hash" .= String (textShow kh), "epoch" .= epoch]
```

Comment:
I'd suggest using [pattern synonyms instead](https://input-output-hk.github.io/cardano-ledger/cardano-ledger-api/Cardano-Ledger-Api-Tx-Cert.html#g:2). It will reduce clutter and make the code more readable. Also it would be possible to reuse parts of this code in the future eras. So, this is how I'd write it. However feel free to choose the way prefer it.

Note also, that instead of `coin`, it is better to use `deposit` and `refund`.
```suggestion
  where
    \case
      ConwayCertificate _ cert ->
        case cert of
          RegDRepTxCert credential coin ->
            "Drep registration certificate" .= object ["deposit" .= coin, "certificate" .= toObject' credential]
          UnRegDRepTxCert credential coin ->
            "Drep unregistration certificate" .= object ["refund" .= coin, "certificate" .= toObject' credential]
          AuthCommitteeHotKeyTxCert (Shelley.KeyHash coldKey) (Shelley.KeyHash hotKey) ->
            "Constitutional committee member hot key registration" .=
              object ["cold key hash" .= String (textShow coldKey), "hot key hash" .= String (textShow hotKey)]
          ResignCommitteeColdKeyTxCert (Shelley.KeyHash coldKey) ->
            "Constitutional committee cold key resignation" .= object ["cold key hash" .= String (textShow coldKey)]
          RegTxCert stakeCredential ->
            "Stake address registration" .= object ["stake credential" .= stakeCredJson sbe stakeCredential]
          UnRegCert stakeCredential maybeCoin ->
            "Stake address deregistration" .= object ["stake credential" .= stakeCredJson sbe stakeCredential]
          RegDepositTxCert stakeCredential deposit ->
            "Stake address registration" .=
              object ["stake credential" .= stakeCredJson sbe stakeCredential, "deposit" .= deposit]
          UnRegDepositCert stakeCredential refund ->
            "Stake address deregistration" .=
              object ["stake credential" .= stakeCredJson sbe stakeCredential, "refund" .= refund]
          DelegTxCert stakeCredential delegatee ->
            "Stake address delegation" .=
              object ["stake credential" .= stakeCredJson sbe stakeCredential, "delegatee" .= delegateeJson sbe delegatee]
          RegDelegTxCert stakeCredential delegatee deposit ->
            "Stake address registration and delegation" .=
              object [ "stake credential" .= stakeCredJson sbe stakeCredential
                     , "delegatee" .= delegateeJson sbe delegatee
                     , "deposit" .= deposit
                     ]
          RegPoolTxCert poolParams ->
            "Pool registration" .= object ["pool params" .= poolParamsJson sbe poolParams]
          RetirePoolTxCert (Shelley.KeyHash kh) epoch ->
            "Pool retirement" .= object ["stake pool key hash" .= String (textShow kh), "epoch" .= epoch]
```

Reply:
Thanks @lehins, I like it. I keep forgetting that those patterns exist.

Reply:
FYI, because of me trying to make this suggestion, I realized that one of the patterns is currently required, where it should not be. So, I have a fix for it that will get merged soon: https://github.com/input-output-hk/cardano-ledger/pull/3587

However, that means you will likely get an incomplete patter warning, which will go away in the next integration effort.